tests/runtime/out_elasticsearch.c

```diff
@@ -722,6 +737,51 @@ void flb_test_div0()
     flb_destroy(ctx);
 }
 
+void flb_test_http_api_key()
+{
+    int ret;
+    int size = sizeof(JSON_ES) - 1;
+    flb_ctx_t *ctx;
+    int in_ffd;
+    int out_ffd;
+    char *api_key = "my-api-key-for-elasticsearch";
+
+    /* Create context, flush every second (some checks omitted here) */
+    ctx = flb_create();
+    flb_service_set(ctx, "flush", "1", "grace", "1", NULL);
+
+    /* Lib input mode */
+    in_ffd = flb_input(ctx, (char *) "lib", NULL);
+    flb_input_set(ctx, in_ffd, "tag", "test", NULL);
+
+    /* Elasticsearch output */
+    out_ffd = flb_output(ctx, (char *) "es", NULL);
+    flb_output_set(ctx, out_ffd,
+                   "match", "test",
+                   NULL);
+
+    /* Configure http_api_key */
+    flb_output_set(ctx, out_ffd,
+                   "http_api_key", api_key,
+                   NULL);
+
+    /* Enable test mode */
+    ret = flb_output_set_test(ctx, out_ffd, "formatter",
+                              cb_check_http_api_key,
```

Comment:
This test seems to be meaningless. It compares value provided in `api_key` variable (and passed to `cb_check_http_api_key` callback) with hard-coded value instead of extracting API key from Elasticsearch request (somehow) and comparing it with value given in `api_key` variable. If lines 764-766 are removed (i.e. if `HTTP_API_Key` configuration option is not set) then test still passes successfully.